src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -928,7 +928,9 @@ internal bool MethodIsCompatibleWithDelegateOrFunctionPointer(BoundExpression? r
                     return true;
                 }
 
-                return targetKind == TypeKind.FunctionPointer && ConversionsBase.HasImplicitPointerConversion(source, destination);
+                return targetKind == TypeKind.FunctionPointer
```

Comment:
>FunctionPointer [](start = 46, length = 15)

[By spec](https://github.com/dotnet/csharplang/blob/master/proposals/function-pointers.md#function-pointer-conversions), the pointer to void conversion should be permitted even for non-function-pointers.